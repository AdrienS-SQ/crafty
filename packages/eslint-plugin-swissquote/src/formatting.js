const { warn } = require("./utils");

// Command line formatting checker
// Mainly the formatting rules and
// basic mistakes catching
module.exports = {
  env: {
    browser: true,
    node: false,
    amd: true
  },
  rules: {
    // Code Style
    "arrow-spacing": [warn(), { before: true, after: true }],
    "brace-style": [warn(), "1tbs"],
    "comma-dangle": [warn(), "never"],
    "comma-spacing": [warn(), { before: false, after: true }],
    // "consistent-this": [warn(), "that"],
    // "func-names": warn(),
    // "func-style": warn(),
    "generator-star-spacing": ["error", { before: false, after: true }],
    indent: [
      warn(),
      2,
      { SwitchCase: 1, ignoredNodes: ["ConditionalExpression"] }
    ],
    "jsx-quotes": [warn(), "prefer-double"],
    "keyword-spacing": warn(),
    // "max-depth": [warn(), 4],
    // "max-len": [warn(), 80, 4],
    // "max-nested-callbacks": [warn(), 2],
    // "max-params": [warn(), 3],
    // "max-statements": [warn(), 10],
    "new-cap": [warn(), { newIsCapExceptions: ["default"] }],
    "new-parens": warn(),
    "no-array-constructor": warn(),
    "no-bitwise": warn(),
    // "no-div-regex": warn(),
    // "no-else-return": warn(),
    "no-empty": warn(),
    "no-empty-character-class": warn(),
    "no-extra-parens": [
      warn(),
      "all",
      {
        nestedBinaryExpressions: false,
        conditionalAssign: false,
        ignoreJSX: "multi-line"
      }
    ],
    "no-extra-semi": warn(),
    "no-floating-decimal": warn(),
    "no-lone-blocks": warn(),
    // "no-mixed-requires": [warn(), false],
    "no-multi-str": warn(),
    "no-multi-spaces": warn(),
    "no-negated-condition": warn(),
    "no-nested-ternary": warn(),
    "no-new-object": warn(),
    "no-new-wrappers": warn(),
    // "no-plusplus": 0,
    "no-self-compare": warn(),
    "no-shadow-restricted-names": warn(),
    "no-spaced-func": warn(),
    "no-sparse-arrays": warn(),
    "no-mixed-spaces-and-tabs": warn(),
    "no-trailing-spaces": warn(),
    "no-useless-concat": warn(),
    //"no-useless-escape": warn(),
    "no-void": warn(),
    "no-with": warn(),
    // "one-var": warn(),
    // "quote-props": warn(),
    quotes: [warn(), "double", "avoid-escape"],
    "rest-spread-spacing": ["error", "never"],
    semi: warn(),
    // "sort-vars": warn(),
    "space-infix-ops": warn(),
    "space-unary-ops": [warn(), { words: true, nonwords: false }],
    // "spaced-comment": [warn(), "always", { "exceptions": ["-", "+", "*"] }],
    "template-curly-spacing": warn(),
    // "wrap-regex": warn(),
    "yield-star-spacing": [warn(), "after"]
  }
};
