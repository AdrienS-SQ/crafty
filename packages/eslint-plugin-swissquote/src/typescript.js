const { warn } = require("./utils");

// EcmaScript 6 specific configuration
module.exports = {
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: "module",
    ecmaFeatures: {
      jsx: true
    },

    // typescript-eslint specific options
    warnOnUnsupportedTypeScriptVersion: true
  },
  rules: {
    "no-array-constructor": "off",
    "@swissquote/swissquote/@typescript-eslint/no-array-constructor": warn(),

    "@swissquote/swissquote/@typescript-eslint/type-annotation-spacing": warn(),

    indent: "off",
    "@swissquote/swissquote/@typescript-eslint/indent": [warn(), 2],

    semi: "off",
    // TODO :: enable this on next release of eslint rules
    //"@swissquote/swissquote/@typescript-eslint/semi": [warn()],

    "no-extra-parens": "off",
    "@swissquote/swissquote/@typescript-eslint/no-extra-parens": [
      warn(),
      "all",
      {
        nestedBinaryExpressions: false,
        conditionalAssign: false,
        ignoreJSX: "multi-line"
      }
    ]
  }
};
