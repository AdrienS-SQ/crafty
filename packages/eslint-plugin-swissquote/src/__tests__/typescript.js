const { prepareCLIEngine, lint } = require("../../test_utils");
const engine = prepareCLIEngine(
  require("../formatting"),
  require("../typescript"),
  { parser: require.resolve("@typescript-eslint/parser") }
);

it("Fails on badly formatted TypeScript code", () => {
  const result = lint(
    engine,
    `
module.exports = function initJS  (gulp, config: {}, watchers): string[] {
  const js = config.js,
     jsTasks: string[] = [];

  for (const name in js) {
    if (!js.hasOwnProperty(name)) {
      continue;
      }

    const taskName = \`js_\${name}\`;

    if (!compileWithWebpack(js[name])) {
      gulp.task(taskName, jsTaskES5(gulp, config, watchers, js[name]));
        watchers.add(js[name].watch || js[name].source, taskName);
    }

    jsTasks.push( taskName );
  }

  gulp.task("js", jsTasks);

  return ["js"];
};
`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toBe(0);
  expect(result.errorCount).toBe(4);
});
