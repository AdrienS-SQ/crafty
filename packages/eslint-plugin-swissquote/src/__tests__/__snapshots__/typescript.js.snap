// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fails on badly formatted TypeScript code 1`] = `
Array [
  Object {
    "column": 35,
    "fix": Object {
      "range": Array [
        33,
        35,
      ],
      "text": " ",
    },
    "line": 2,
    "message": "Multiple spaces found before '('.",
    "nodeType": "Punctuator",
    "ruleId": "no-multi-spaces",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 4,
    "fix": Object {
      "range": Array [
        100,
        105,
      ],
      "text": "    ",
    },
    "line": 4,
    "message": "Expected indentation of 4 spaces but found 5.",
    "messageId": "wrongIndentation",
    "nodeType": "Identifier",
    "ruleId": "@swissquote/swissquote/@typescript-eslint/indent",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 7,
    "endLine": 9,
    "fix": Object {
      "range": Array [
        209,
        215,
      ],
      "text": "    ",
    },
    "line": 9,
    "message": "Expected indentation of 4 spaces but found 6.",
    "messageId": "wrongIndentation",
    "nodeType": "Punctuator",
    "ruleId": "@swissquote/swissquote/@typescript-eslint/indent",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 9,
    "endLine": 15,
    "fix": Object {
      "range": Array [
        367,
        375,
      ],
      "text": "      ",
    },
    "line": 15,
    "message": "Expected indentation of 6 spaces but found 8.",
    "messageId": "wrongIndentation",
    "nodeType": "Identifier",
    "ruleId": "@swissquote/swissquote/@typescript-eslint/indent",
    "severity": 2,
  },
]
`;
